content/blog/2022/10/envvar-policy.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Enforcing compliance in your environment variables
+authors:
+- José Guilherme Vanz
+date: 2022-11-01
+---
+
+We're glad to present to Kubewarden users the new [environment-variable-policy](https://github.com/kubewarden/environment-variable-policy) policy.
+It's able to inspect containers, init containers, and ephemeral containers, and restricts
+their usage of environment variables by checking their names and values defined under the
+containers' [`env[*]`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#envvar-v1-core).
+As always, the policy can be found in [Github](https://github.com/kubewarden/environment-variable-policy)
+and all the  artifacts, including the BOM files, are signed with Sigstore
+
+### What is so useful about the new environment-variable policy?
+
+The recent added policy https://github.com/kubewarden/env-variable-secrets-scanner-policy allow users
+to check if there are some leaked sensitive data in the environment variables. However, until the
+environment-variable-policy there were no way to check other environment variables aspects.
+This is fixed now. In this new policy users can validate which variables name and values their
+resources should have. For that, the policy gives 4 rule operator to control the
+environment variable defined in the containers. They are:
+
+- `anyIn`: Works as a deny list of your usage of volumes. It checks if any of the `environmentVariables` are in the Pod/Workload resource:
+    ```yaml
+settings:
+  rules:
+    - reject: anyIn
+      environmentVariables:
+        - name: "envvar1"
+        - name: "envvar2"
+    ```
+- `anyNotIn`: Works as an allow list. Checks if any of the `environmentVariables` are not in the Pod/Workload resource:
+    ```yaml
+settings:
+  rules:
+    - reject: anyNotIn
+      environmentVariables:
+        - name: "envvar2"
+          value: "envvar2_value"
+        - name: "envvar3"
+    ```
+- `allAreUsed`: The container cannot use all listed volumes at once. Checks if all of the `environmentVariables` are in the Pod/Workload resource:
+    ```yaml
+settings:
+  rules:
+    - reject: allAreUsed
+      environmentVariables:
+        - name: "envvar2"
+          value: ""
```

Comment:
what is the behaviour when the value is `""`?

Reply:
I don't like to explain this operator by using just one value, which is also a corner case. I would use something more "traditional", like it has been done inside of the other values.

I also think it would be worth to mention both here **and** in the policy README + metadata how `""` is being handled.